preserveData.js

```diff
@@ -0,0 +1,31 @@
+const storeData = {};
+const nameData = document.getElementById('name');
+const emailData = document.getElementById('mail');
+const messageData = document.getElementById('msg');
+
+function saveToLocal(storeData) {
+  localStorage.setItem('info', JSON.stringify(storeData));
+}
+
+nameData.addEventListener('change', (event) => {
+  storeData.name = event.target.value;
+  saveToLocal(storeData);
+});
+
+emailData.addEventListener('change', (event) => {
+  storeData.mail = event.target.value;
+  saveToLocal(storeData);
+});
+
+messageData.addEventListener('change', (event) => {
+  storeData.msg = event.target.value;
+  saveToLocal(storeData);
+});
```

Comment:
- [ ] Kindly fix the textarea field, as i6 currently is displaying `undefined` and is unable to store user input
![image](https://user-images.githubusercontent.com/81932478/216794642-a649d67b-da2d-444b-9d20-475d2d58d00d.png)

Reply:
Hello @Munsa1 ! I tested it several times and it didn't show me an error, do you have any idea how the error is being generated? Since it doesn't appear for me, I don't know where to start looking, thanks for the feedback!
![Screenshot (28)](https://user-images.githubusercontent.com/116699473/216795198-d3f7a89d-2dfc-45b2-941c-3f483178213a.png)